src/client/auth/scram.rs

```diff
@@ -57,19 +57,32 @@ struct CacheEntry {
 }
 
 /// The versions of SCRAM supported by the driver (classified according to hash function used).
-#[derive(Hash, Eq, PartialEq, Clone)]
+#[derive(Debug, Hash, Eq, PartialEq, Clone)]
 pub(crate) enum ScramVersion {
     Sha1,
     Sha256,
 }
 
+/// Contains the authentication info derived from a Credential.
+pub(crate) struct ClientAuthInfo<'a> {
```

Comment:
I wanted to abstract out the code that parses and validates these fields, and having a struct seemed better than a 3-tuple